src/controllers/productController/productController.ts

```diff
@@ -43,23 +42,23 @@ class ProductController {
       }
 
       //const UserService = new userService();
-      const user = await UserService.getUserById(userId) as any;
+      const user = await User.findByPk(userId) as any;
       const userRole = await Role.findByPk(user.dataValues.roleId) as any;
 
       if (!user || userRole.dataValues.name !== 'seller') {
         res.status(403).json({ message: 'Only sellers can create products' });
         return;
       }
 
-      //const productService = new ProductService();
-      const existingProduct = await productService.getProductByNameAndSellerId(productDetails.productName, productDetails.userId);
+      
+      const existingProduct = await productService.getProductByNameAndSellerId(productDetails.productName, userId);
       if (existingProduct) {
         res.status(409).json({ message: 'Product already exists. Consider updating stock levels instead.'});
         return;
       }
 
       const createdProduct = await Product.create({
-        userId: req.body.userId,
+        userId: user.dataValues.id,
```

Comment:
You can use the middleware called sellerAuth ti get the id and detqils of the logged in user